webapp/src/components/boardsSwitcher/boardsSwitcher.tsx

```diff
@@ -93,20 +109,70 @@ const BoardsSwitcher = (props: Props): JSX.Element => {
             </div>
             {shouldViewSearchForBoardsTour && <div><SearchForBoardsTourStep/></div>}
             {
-                Utils.isFocalboardPlugin() &&
-                <IconButton
-                    size='small'
-                    inverted={true}
-                    className='add-board-icon'
-                    onClick={props.onBoardTemplateSelectorOpen}
-                    icon={<AddIcon/>}
-                />
+                Utils.isFocalboardPlugin() && (
+                    <MenuWrapper>
+                        <IconButton
+                            size='small'
+                            inverted={true}
+                            className='add-board-icon'
+                            icon={<AddIcon/>}
+                        />
+                        <Menu>
+                            <Menu.Text
+                                id='create-new-board-option'
+                                icon={<CompassIcon icon='plus' />}
+                                onClick={props.onBoardTemplateSelectorOpen}
+                                name='Create new board'
+                            />
+                            <Menu.Text
+                                id='createNewCategory'
+                                name={intl.formatMessage({id: 'SidebarCategories.CategoryMenu.CreateNew', defaultMessage: 'Create New Category'})}
+                                icon={
+                                    <CompassIcon
+                                        icon='folder-plus-outline'
+                                        className='CreateNewFolderIcon'
+                                    />
+                                }
+                                onClick={handleCreateNewCategory}
+                            />
+                        </Menu>
+                    </MenuWrapper>
+                )
             }
 
             {
                 showSwitcher &&
                 <BoardSwitcherDialog onClose={() => setShowSwitcher(false)} />
             }
+
+            {
+                showCreateCategoryModal && (
+                    <CreateCategory
```

Comment:
This is duplicated code. Can you try to move the logic inside the component itself? I can help with this.

Reply:
Sure.